src/coreclr/vm/assembly.cpp

```diff
@@ -1392,7 +1452,9 @@ INT32 Assembly::ExecuteMainMethod(PTRARRAYREF *stringArgs, BOOL waitForOtherThre
 
             hr = RunMain(pMeth, 1, &iRetVal, stringArgs);
 
+#if !defined(TARGET_BROWSER)
```

Comment:
Why is this defined away? This should be a nop on system where Objective-C isn't enabled. Was this causing issues?

Reply:
Is this because `RunMain` returns without exiting on browser? If so, that deserves a comment. Just defining it away is confusing.

Reply:
Yes, we want the thread to keep "attached". I will add comment